src/main/java/frc/robot/commands/CoralGround.java

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import java.util.function.BooleanSupplier;
+
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.subsystems.CoralGroundIntake;
+import frc.robot.subsystems.CoralGroundIntake.CoralGroundIntakeState;
+
+/* You should consider using the more terse Command factories API instead https://docs.wpilib.org/en/stable/docs/software/commandbased/organizing-command-based.html#defining-commands */
+public class CoralGround extends Command {
+  /** Creates a new scoreL1. */
+  private final CoralGroundIntake m_coralGroundIntake;
+  private final BooleanSupplier m_endEffectorHasCoral;
+  private boolean m_finished = false;
+  private final Timer m_timer = new Timer();
+
+  public CoralGround(CoralGroundIntake CoralGroundIntake, BooleanSupplier hasCoral) {
+    // Use addRequirements() here to declare subsystem dependencies.
+    m_endEffectorHasCoral = hasCoral;
+    m_coralGroundIntake = CoralGroundIntake;
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    if (m_endEffectorHasCoral.getAsBoolean()) {m_finished = true; return;}
+    m_finished = false; 
+    System.out.println("Run! deploy angle intake thingey");
+    SmartDashboard.putBoolean("coralGroundIntake/running", true);
+    if (m_coralGroundIntake.m_state == CoralGroundIntakeState.SCORE_ANGLE) {
```

Comment:
See other comment: the subsystem needs to have a getter for the State. Grab it once into a local variable.